src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/RoleAssignmentService.java

```diff
@@ -37,17 +41,29 @@ public RoleAssignments getRoleAssignments(String userId) {
 
     public List<String> getCaseReferencesForAGivenUser(String userId) {
         final RoleAssignments roleAssignments = this.getRoleAssignments(userId);
+        return getValidCaseIds(roleAssignments.getRoleAssignments());
+    }
 
-        return roleAssignments.getRoleAssignments().stream()
-            .filter(roleAssignment -> isAValidRoleAssignments(roleAssignment))
+    public List<String> getCaseReferencesForAGivenUser(String userId, CaseTypeDefinition caseTypeDefinition) {
+
+        final RoleAssignments roleAssignments = this.getRoleAssignments(userId);
+        List<RoleAssignment> filteredRoleAssignments = roleAssignmentsFilteringService
+                .filter(roleAssignments, caseTypeDefinition);
+
+        return getValidCaseIds(filteredRoleAssignments);
+    }
+
+    private List<String> getValidCaseIds(List<RoleAssignment> roleAssignmentsList) {
+        return roleAssignmentsList.stream()
+            .filter(this::isAValidRoleAssignments)
             .map(roleAssignment -> roleAssignment.getAttributes().getCaseId())
             .flatMap(Optional::stream)
             .collect(Collectors.toList());
     }
 
     private boolean isAValidRoleAssignments(RoleAssignment roleAssignment) {
```

Comment:
may be we can use isValidRoleAssignments

Reply:
not sure what to do in here